src/makeInfinityAdder.js

```diff
@@ -1,10 +1,27 @@
 'use strict';
 
 /***
+ *
  * @return {function}
  */
+
 function makeInfinityAdder() {
-  // write code here
+  let result = 0;
+
+  const adder = (arg) => {
+    if (arg === undefined) {
+      const sum = result;
+      result = 0;
+
+      return sum;
+    }
+
+    result += arg;
+
+    return adder;
+  };
```

Comment:
Consider adding a check to ensure that the argument passed to `adder` is a number. This will make your function more robust by preventing unexpected behavior when non-numeric values are added.